docs/Personalities.md

```diff
@@ -0,0 +1,276 @@
+hash are assumed equal, even if the full Merkle leaf structure differs
+(e.g. because it has a different timestamp attached).
+
+This means that a personality which wants to prevent duplicates entries can
+do so by setting the `LeafIdentityHash` value on new Merkle tree leaves
+appropriately. The personality code also needs to cope with duplicates: the
+Merkle tree leaf in a successful `QueueLeaves` response may not be the same
+as the tree leaf in the corresponding `QueueLeaves` request (it will have the
+same `LeafIdentityHash` but may differ in fields that are not covered by this
+hash, e.g. it may have an earlier timestamp).
+
+
+### Auditability
+
+If the personality and the Trillian core services are maintained by different
+operators, then there may be an implicit trust boundary between the two at the
+Trillian gRPC interface.
+
+If this is the case, then the personality may need to be responsible for
+storing data that allows auditing of that trust boundary.
+
+For example, if an external monitor detects that a signed tree head is not
+consistent with an earlier signed tree head, is there enough information
+available to determine whether this is a problem with the personality or with
+the core Trillian service?
+
+If the personality maintains a store of the signed log roots provided by the
+Trillian core, it can then use this to audit failure cases and assign blame
+appropriately &ndash; effectively acting as a monitor for the Trillian core
+service.
+
+
+## Practical Responsibilities
+
+### External API
+
+Some transparency scenarios may have the personality code embedded into
+another application, but in many environments the personality provides an
+externally-visible API.
+
+The design of such an API depends on the application, but key points to
+consider include:
+
+ - **syntax** &ndash; the mechanism used to access the API, where possibilities
+   include:
+   - HTTP(S) + JSON: A web-based interface has the advantage of being
+     accessible using standard tools, ranging from
+     [standard](https://golang.org/pkg/net/http/)
+     [libraries](https://docs.python.org/3/library/http.client.html) to
+     command-line tools like `curl` and `wget`.  (The Certificate Transparency
+     API described in [RFC 6962](https://tools.ietf.org/html/rfc6962) takes
+     this approach.)
+   - gRPC: An RPC-based interface allows for additional tooling (such as
+     mechanical checks of schema adherence) and may allow features that are not
+     easily achieved in a RESTful web API (such as
+     [continuous streaming](https://grpc.io/docs/guides/concepts.html#server-streaming-rpc)
+     of data).
+
+ - **semantics** &ndash; the set of primitives and data structures exposed by
+   the API, for example:
+   - What submission formats are accepted?
+   - What cryptographic promises/guarantees does the personality generate?
+   - Which Merkle tree primitives are exposed to external users?
```

Comment:
Aside: We should document the cryptographic guarantees that a personality can provide, and what you need to do to provide them.